common/src/seg_tree.c

```diff
@@ -253,6 +305,26 @@ struct seg_tree_node* seg_tree_find_nolock(
     return NULL;
 }
 
+
+/*
+ * Search tree for an entry that overlaps with given range of [start, end].
```

Comment:
I think it makes more sense to have this comment on the nolock function, since it is the one that actually implements the searc, and have this comment be the one that says "this is the same as seg_tree_find_nolock, but adds locking".